src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -56,19 +66,51 @@ class Lambda {
     dotcomSurrogateKey
   }
 
-  private def sendFastlyPurgeRequestAndAmpPingRequest(contentId: String, purgeType: PurgeType, serviceId: String, surrogateKey: String, fastlyApiKey: String): Boolean = {
-    if (sendFastlyPurgeRequest(contentId, purgeType, serviceId, surrogateKey, fastlyApiKey))
-      sendAmpPingRequest(contentId)
-    else
-      false
+  // Left[Id] for Delete, Right[Id] for Update
+  def decidePurgeType: PartialFunction[Event, Either[Id, Id]] = {
+    case Event(_, EventType.Update, _, _, Some(EventPayload.Content(content))) => Right(content.id)
+    case Event(_, EventType.RetrievableUpdate, _, _, Some(EventPayload.RetrievableContent(content))) => Right(content.id)
+    case Event(_, EventType.Delete, _, _, Some(EventPayload.RetrievableContent(content))) => Left(content.id)
+  }
+
+  def purgeDelete: Id => Future[List[PurgeResult]] = { id =>
+    val fastlyResult: Future[Boolean] = sendFastlyPurgeRequest(id, Hard, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(id), config.fastlyDotcomApiKey)
+    val ampResult: Future[Boolean] = fastlyResult.flatMap(if (_) sendAmpPingRequest(id) else Future.successful(false))
+    Future.sequence(fastlyResult.map(PurgeResult(_, Dotcom)) :: ampResult.map(PurgeResult(_, AMP)) :: Nil)
+  }
+
+  def purgeUpdate: Id => Future[List[PurgeResult]] = { id =>
```

Comment:
Here I'm assuming the behaviour should be the same for Update and RetrievableUpdate event types, i.e. we should purge Dotcom, Nextgen and Mapi.
Also I've assumed the form of the requests should be the same - I wasn't sure why we added a .json to the payloadId for Update events and not for RetrievableUpdate events

Reply:
The `.json` should be kept for the `fastlyApiNextgenServiceId ` because the path of the resource being cleared ends with `.json`

Reply:
Thanks for this! I've fixed this with the latest commit